src/datasets/processors/subsample/script.py

```diff
@@ -56,31 +71,76 @@
     choice_probs = [ probs[batch] for batch in choice_batch ]
     obs_index = np.random.choice(choice_ix, size=n_obs, replace=False, p=choice_probs)
 else:
-    obs_index = np.random.choice(np.where(obs_filt)[0], n_vars, replace=False)
+    obs_index = np.random.choice(np.where(obs_filt)[0], n_obs, replace=False)
+
+# subsample obs
+adata_output = adata_input[obs_index].copy()
+if par["input_mod2"] is not None:
+    adata_output_mod2 = adata_mod2[obs_index].copy()
+
+# filter cells and genes
+if par["input_mod2"] is not None:
+    n_cells =  scprep.utils.toarray(adata_output.X.sum(axis=1)).flatten()
+    n_cells_mod2 =  scprep.utils.toarray(adata_output_mod2.X.sum(axis=1)).flatten()
```

Comment:
It seems a bit redundant to install scprep just for this.

```suggestion
    n_cells =  adata_output.X.sum(axis=1).A.flatten()
    n_cells_mod2 =  adata_output_mod2.X.sum(axis=1).A.flatten()
```